Sources/Networking/HTTPClient/ETagManager.swift

```diff
@@ -70,8 +81,11 @@ class ETagManager {
         }
 
         if self.shouldUseCachedVersion(responseCode: statusCode) {
-            if let storedResponse = self.storedHTTPResponse(for: request, withRequestDate: response.requestDate) {
-                return storedResponse
+            if let storedResponse = self.storedETagAndResponse(for: request) {
+                let newResponse = storedResponse.withUpdatedDate()
+
+                self.store(newResponse, for: request)
+                return newResponse.asResponse(withRequestDate: response.requestDate)
```

Comment:
I am a bit confused about this change. If I understand ti correctly, it is creating a new response with a date set to now, then saving that, then returning it with the request date of the request?

Reply:
Oh I see.. It saves the last used, then returns the response with the updated request date

Reply:
agreed. updating